### PR TITLE
kibana: added authlog* index pattern

### PR DESCRIPTION
**Pull request by pettersv**
*Issue #, if available:* #140

*Description of changes:*
Added the authlog* index pattern

*Test Results:*

**Note: If this PR is related to Helm, please also update the README for related documentation changes. Thanks.**
**https://github.com/opendistro-for-elasticsearch/opendistro-build/blob/master/helm/README.md**

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by OlleLarsson** [comment]
Okay, this looks fine altough you still have to update [curator](https://github.com/elastisys/opendistro-build/blob/main/helm/opendistro-es/files/action_file.yml) to remove old indices.
I'd also like you to not use the commit prefix `kibana:` as these changes are not onlt releated to kibana :)

File: helm/opendistro-es/files/action_file.yml [hunk 1]
```diff
@@ -108,3 +108,38 @@ actions:
       unit_count: {{ .Values.curator.retention.kubernetes_days }}
     - filtertype: kibana
       exclude: True
+  7:
```
Comment (OlleLarsson) [inline helm/opendistro-es/files/action_file.yml:111]: Can you update the numbers [here](https://github.com/elastisys/opendistro-build/blob/main/helm/opendistro-es/files/postgresql-actions.yml) to 9 and 10?

Comment (pettersv) [reply helm/opendistro-es/files/action_file.yml:111]: sure

File: helm/opendistro-es/files/action_file.yml [hunk 2]
```diff
@@ -108,3 +108,38 @@ actions:
       unit_count: {{ .Values.curator.retention.kubernetes_days }}
     - filtertype: kibana
       exclude: True
+  9:
+    action: delete_indices
+    description: "Clean up the oldest authlog-* indices that exceeds total disk space {{ .Values.curator.retention.kubernetes_gb }} GB"
```
Comment (OlleLarsson) [inline helm/opendistro-es/files/action_file.yml]:
```suggestion
    description: "Clean up the oldest authlog-* indices that exceeds total disk space {{ .Values.curator.retention.authlog_gb }} GB"
```

File: helm/opendistro-es/files/action_file.yml [hunk 3]
```diff
@@ -108,3 +108,38 @@ actions:
...
+    options:
+      continue_if_exception: False
+      ignore_empty_list: True
+      allow_ilm_indices: True
+    filters:
+    - filtertype: pattern
+      kind: regex
+      value: 'authlog-*'
+    - filtertype: space
+      disk_space: {{ .Values.curator.retention.kubernetes_gb }}
```
Comment (OlleLarsson) [inline helm/opendistro-es/files/action_file.yml]:
```suggestion
      disk_space: {{ .Values.curator.retention.authlog_gb }}
```

File: helm/opendistro-es/files/action_file.yml [hunk 4]
```diff
@@ -108,3 +108,38 @@ actions:
...
+      use_age: True
+      source: creation_date
+    - filtertype: kibana
+      exclude: True
+  10:
+    action: delete_indices
+    description: "Clean up the authlog-* indices that are older then {{ .Values.curator.retention.kubernetes_days }} days"
```
Comment (OlleLarsson) [inline helm/opendistro-es/files/action_file.yml]:
```suggestion
    description: "Clean up the authlog-* indices that are older then {{ .Values.curator.retention..authlog_days }} days"
```

File: helm/opendistro-es/files/action_file.yml [hunk 5]
```diff
@@ -108,3 +108,38 @@ actions:
...
+      allow_ilm_indices: True
+    filters:
+    - filtertype: pattern
+      kind: regex
+      value: 'authlog-*'
+    - filtertype: age
+      source: creation_date
+      direction: older
+      unit: days
+      unit_count: {{ .Values.curator.retention.kubernetes_days }}
```
Comment (OlleLarsson) [inline helm/opendistro-es/files/action_file.yml]:
```suggestion
      unit_count: {{ .Values.curator.retention.authlog_days }}
```

**Review by OlleLarsson — COMMENTED** [review]
Almost there I think but we should add separate variables for configuring removal of the authlog indices.

File: helm/opendistro-es/files/action_file.yml [hunk 6]
```diff
@@ -108,3 +108,38 @@ actions:
       unit_count: {{ .Values.curator.retention.kubernetes_days }}
     - filtertype: kibana
       exclude: True
+  9:
```
Comment (OlleLarsson) [inline helm/opendistro-es/files/action_file.yml]:
```suggestion
  7:
```

File: helm/opendistro-es/files/action_file.yml [hunk 7]
```diff
@@ -108,3 +108,38 @@ actions:
...
+    - filtertype: pattern
+      kind: regex
+      value: 'authlog-*'
+    - filtertype: space
+      disk_space: {{ .Values.curator.retention.authlog_gb }}
+      use_age: True
+      source: creation_date
+    - filtertype: kibana
+      exclude: True
+  10:
```
Comment (OlleLarsson) [inline helm/opendistro-es/files/action_file.yml]:
```suggestion
  8:
```

**Review by OlleLarsson — COMMENTED** [review]
I realize that one of my previous comments was badly formulated and did not come across correctly, but I mean that you should update the numbers [here](https://github.com/elastisys/opendistro-build/blob/main/helm/opendistro-es/files/postgresql-actions.yml) to 9 and 10 :)

Other than that I think this one is good to go.

**Comment by OlleLarsson** [comment]
Ok, I think this one is good to go.
Just remember to that you'll have to increase [chart version](https://github.com/elastisys/opendistro-build/blob/main/helm/opendistro-es/Chart.yaml#L29) before packaging it.

**Review by OlleLarsson — APPROVED** [review]

**Pull request merged**